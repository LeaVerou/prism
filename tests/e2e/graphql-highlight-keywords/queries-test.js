const { expect } = require('../../helper/chai');
const prismFactory = require('../../helper/prism-loader');

const Prism = prismFactory.createInstance(['graphql', 'highlight-keywords']);

describe('E2E test - Prism#highlights queries correctly', () => {
	it('query ex. 1', () => {
		expect(subject(`
query getProductById($id: ID!) {
	product(id: $id) {
		title
		handle
		createdAt
	}
}
`)).toMatchSnapshot();
	});

	it('query shorthand ex. 1', () => {
		expect(subject(`
{
	orders(id: 4) {
		edges {
			node {
				# Order fields
				# Query fields
			}
			cursor
		}
		pageInfo {
			hasNextPage
			hasPreviousPage
		}
	}
}
`)).toMatchSnapshot();
	});
});

function subject(text) {
	return Prism.highlight(text, Prism.languages.graphql, 'graphql');
}
