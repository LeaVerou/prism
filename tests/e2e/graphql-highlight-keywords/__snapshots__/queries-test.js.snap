// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`E2E test - Prism#highlights queries correctly query ex. 1 1`] = `
<span class="token keyword keyword-query">
  query
</span>
<span class="token definition-query function">
  getProductById
</span>
<span class="token punctuation">
  (
</span>
<span class="token variable">
  $id
</span>
<span class="token punctuation">
  :
</span>
<span class="token scalar">
  ID
</span>
<span class="token operator">
  !
</span>
<span class="token punctuation">
  )
</span>
<span class="token punctuation">
  {
</span>
<span class="token property-query">
  product
</span>
<span class="token punctuation">
  (
</span>
<span class="token attr-name">
  id
</span>
<span class="token punctuation">
  :
</span>
<span class="token variable">
  $id
</span>
<span class="token punctuation">
  )
</span>
<span class="token punctuation">
  {
</span>
<span class="token property">
  title
</span>
<span class="token property">
  handle
</span>
<span class="token property">
  createdAt
</span>
<span class="token punctuation">
  }
</span>
<span class="token punctuation">
  }
</span>
`;

exports[`E2E test - Prism#highlights queries correctly query shorthand ex. 1 1`] = `
<span class="token punctuation">
  {
</span>
<span class="token property-query">
  orders
</span>
<span class="token punctuation">
  (
</span>
<span class="token attr-name">
  id
</span>
<span class="token punctuation">
  :
</span>
<span class="token number">
  4
</span>
<span class="token punctuation">
  )
</span>
<span class="token punctuation">
  {
</span>
<span class="token object">
  edges
</span>
<span class="token punctuation">
  {
</span>
<span class="token object">
  node
</span>
<span class="token punctuation">
  {
</span>
<span class="token comment">
  # Order fields
</span>
<span class="token comment">
  # Query fields
</span>
<span class="token punctuation">
  }
</span>
<span class="token property">
  cursor
</span>
<span class="token punctuation">
  }
</span>
<span class="token object">
  pageInfo
</span>
<span class="token punctuation">
  {
</span>
<span class="token property">
  hasNextPage
</span>
<span class="token property">
  hasPreviousPage
</span>
<span class="token punctuation">
  }
</span>
<span class="token punctuation">
  }
</span>
<span class="token punctuation">
  }
</span>
`;
